### ISSUE TITLE
Source file for entity /tslib.ts not found, check your 'entitiesDirsTs' option with set of entities supplied in config.

### ISSUE THREAD

**Issue by jsprw** [mikro-orm/mikro-orm#194]
**Describe the bug**
1) Can't load entities. Inputted multiple entity classes via the `entities` options and set entitiesDirsTs, entitiesDirs and tsConfigPath. Any idea why it starts loading this tslib.ts?
2) Is it possible to disable the scanning of the directories for the files? It would make distributing the file easier as you can pack it with webpack, this way you can include it with Angular SSR.

**Stack trace**
```[info] MikroORM successfully connected to database xx-dev1 on mysql://xx:*****@xxx.xxx.xxx.xxx:xxxx
[discovery] ORM entity discovery started
[discovery] - processing entity XXEntity
Unhandled Promise rejection: Source file for entity /tslib.ts not found, check your 'entitiesDirsTs' option ; Zone: <root> ; Task: Promise.then ; Value: Error: Source file for entity /tslib.ts not found, check your 'entitiesDirsTs' option
    at TypeScriptMetadataProvider.getSourceFile (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:46:19)
    at TypeScriptMetadataProvider.readTypeFromSource (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:35:24)
    at TypeScriptMetadataProvider.initProperties (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:29:29)
    at TypeScriptMetadataProvider.loadEntityMetadata (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:16:9)
    at MetadataDiscovery.discoverEntity (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\MetadataDiscovery.js:100:9)
    at Function.runSerial (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\utils\Utils.js:199:22)
    at MetadataDiscovery.discover (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\MetadataDiscovery.js:32:13)
    at Function.init (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\MikroORM.js:18:28) Error: Source file for entity /tslib.ts not found, check your 'entitiesDirsTs' option
    at TypeScriptMetadataProvider.getSourceFile (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:46:19)
    at TypeScriptMetadataProvider.readTypeFromSource (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:35:24)
    at TypeScriptMetadataProvider.initProperties (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:29:29)
    at TypeScriptMetadataProvider.loadEntityMetadata (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\TypeScriptMetadataProvider.js:16:9)
    at MetadataDiscovery.discoverEntity (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\MetadataDiscovery.js:100:9)
    at Function.runSerial (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\utils\Utils.js:199:22)
    at MetadataDiscovery.discover (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\metadata\MetadataDiscovery.js:32:13)
    at Function.init (C:\Users\XX\Repositories\XXproj\node_modules\mikro-orm\dist\MikroORM.js:18:28)
[nodemon] clean exit - waiting for changes before restart
```

**To Reproduce**
1. I have no idea how to reproduce this error or why this occurred.

**Expected behavior**
MikroORM starting with the entities supplied in the entities array without scanning directories for more (starting with the supplied entities such as Typeorm does).

**Additional context**
Was planning to switch to MikroORM from Typeorm. MikroORM is used in the backend (same server as Angular SSR). Therefore it would be nice to supply classes instead of scanning directories.

**Versions**

| Dependency | Version |
| - | - |
| node | 12.9.1 |
| typescript | 3.5.3 (needed for Angular) |
| mikro-orm | 3.0.0-alpha.22 |
| your-driver | mysql2 |

**Comment by B4nan** [comment]
You can disable this simply by providing `type` or `entity` option on all decorators. 

Btw the issue is not that ORM tries to load _more_ entities via auto-detection. It just tries to find your TS source files of those entities provided in the `entities` array. And it does that by going thru folders listed in `entitiesDirsTs` (if provided, otherwise in all project files based on `tsconfig.json` configuration). 

But I think your issue is with the bundling, that as probably not supported currently. The path to TS source file of given entity is determined by filename of given JS class. That path is sniffed from decorators. If your JS class is bundled to a single file with other entities, and the file is not named based on the entity class (e.g. `Author` will be in `author.js`, but there are more possible file names), then the lookup of TS source file can't work. 

Maybe the solution would be to allow specifying the path to TS source file of given entity. But again, you can always skip this process by defining the `type` manually (for relationships defining `entity` is also enough). Here is an example of that: https://stackoverflow.com/questions/55523319/how-to-release-or-distribute-an-application-that-uses-mikro-orm

You could also define your own `MetadataProvider` (that is the part of ORM that is causing your problem). 

Can you confirm this happens only when using a bundler like webpack? Did you try to create separate backend project with same configuration? 

Would be great if you could provide example project with this failure. I will be happy to look into this more, but do not have time to set it up myself now. Would love to support bundling too, it might not be that hard if it's just about discovery process and configuration.

**Comment by jsprw** [comment]
Hi, thank you for the quick response!

> You can disable this simply by providing type or entity option on all decorators.
Thank you. Will try this solution tomorrow. Why does it need type or entitity, isn't it possible to read it? How does TypeORM handle types not specified in decorators (e.g. boolean is read without even setting a value)?

> Btw the issue is not that ORM tries to load more entities via auto-detection. It just tries to find your TS source files of those entities provided in the entities array. And it does that by going thru folders listed in entitiesDirsTs (if provided, otherwise in all project files based on tsconfig.json configuration).
Looks like I completely misunderstood the sniffing of files. My bad.

> But I think your issue is with the bundling, that as probably not supported currently. The path to TS source file of given entity is determined by filename of given JS class. That path is sniffed from decorators. If your JS class is bundled to a single file with other entities, and the file is not named based on the entity class (e.g. Author will be in author.js, but there are more possible file names), then the lookup of TS source file can't work.

> Can you confirm this happens only when using a bundler like webpack? Did you try to create separate backend project with same configuration?
Currently I am not bundling anything, I was worrying that the bundling will not work even if I get this to work now. The project is started with the following command: 
`cd src && nodemon -V --watch \"src/api/**/*\" -e ts --exec \"ts-node server.ts\"`

Also I am not using the name tslib nor do I have references to them, that's why I was wondering why it tried to load it..

> You could also define your own MetadataProvider (that is the part of ORM that is causing your problem).
I was trying to avoid this as it would require to write an additional module not maintained by you which may cause incompatibilities after updates in the future.

>Would be great if you could provide example project with this failure. I will be happy to look into this more, but do not have time to set it up myself now. Would love to support bundling too, it might not be that hard if it's just about discovery process and configuration.
I will set up an example project on Github tomorrow.

**Comment by B4nan** [comment]
> Currently I am not bundling anything

Then it might be just an issue with bad tslib autodetection. Would be great if you could `console.log(stack, line, meta.path)` at the end of `Utils.lookupPathFromDecorator()` method. 

> I was trying to avoid this as it would require to write an additional module not maintained by you which may cause incompatibilities after updates in the future.

If we identify the problem as something that can be solved by either modifying current `TypescriptMetadataProvider` or introducing new metadata provider, I will be more than happy to integrate it as part of ORM. 

> I will set up an example project on Github tomorrow.

Perfect, ty!

**Comment by jsprw** [comment]
> Then it might be just an issue with bad tslib autodetection. Would be great if you could console.log(stack, line, meta.path) at the end of Utils.lookupPathFromDecorator() method.

The requested logs, the error occurs for every file imported by entities.

```[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:7:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:9:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:8:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:9:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:12:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:8:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:9:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:9:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:9:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:8:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:15:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:11:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
[
  'Error: ',
  '    at Function.lookupPathFromDecorator (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\utils\\Utils.js:175:26)',
  '    at C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\mikro-orm\\dist\\decorators\\PrimaryKey.js:12:23',
  '    at Object.__decorate (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\tslib\\tslib.js:93:114)',
  '    at Object.<anonymous> (C:\\Users\\XX\\Repositories\\XXXX\\src\\api\\entities\xx_entity(all different but masked).ts:19:5)',
  '    at Module._compile (internal/modules/cjs/loader.js:936:30)',
  '    at Module.m._compile (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:493:23)',
  '    at Module._extensions..js (internal/modules/cjs/loader.js:947:10)',
  '    at Object.require.extensions.<computed> [as .ts] (C:\\Users\\XX\\Repositories\\XXXX\\node_modules\\ts-node\\src\\index.ts:496:12)',
  '    at Module.load (internal/modules/cjs/loader.js:790:32)',
  '    at Function.Module._load (internal/modules/cjs/loader.js:703:12)'
] 3 C:/Users/XX/Repositories/XXXX/node_modules/tslib/tslib.js
```

> If we identify the problem as something that can be solved by either modifying current TypescriptMetadataProvider or introducing new metadata provider, I will be more than happy to integrate it as part of ORM.

If no other solution is found, I will try and contribute a metadata provider. May take a while to see how yours currently works and comparing it to Typeorm's.

**Comment by B4nan** [comment]
Ouch, you are on windows! Now it makes sense. I do normalize the path, but after the tslib check.

**Comment by jsprw** [comment]
Yes... I am... sadly. For this reason, I was worrying about bundling, external paths and webpack don't like each other much, especially cross-platform.

Below is how I dynamically load all the entities in a folder without hardcoding them in the software, might be useful to see in advance how the entities are imported in TypeORM with the possibility to webpack.
 
```
MikroORM.init({
        ....
        entities: await this.getEntities(),
        ....
});
```

```
private static async getEntities(): Promise<any[]> {
        //dynamically load all files in webpack, require.context() does only exist within webpack and tells it to bundle all the **.ts files recursively; this way works way faster than the readdirsync way.
        if (process.env.NODE_ENV === "production") {
            const modules = require.context("../entities", true, /\.ts$/);
            return modules
                .keys()
                .map(r => modules(r))
                .flatMap(name => Object.keys(name).map(k => name[k]));
        } else {
            //therefore this part still exists to test the software locally, it has the same result
            return (await Promise.all(
                fs
                    .readdirSync(join(process.cwd(), "api", "entities"))
                    .map(file => import(`../entities/${file}`)) //import returns a promise, await does not work in a map..
            )).flatMap(name => Object.keys(name).map(k => name[k]));
        }
    }
```

**Comment by B4nan** [comment]
Published `2.7.9` and `3.0.0-alpha.23` with the fix. Feel free to continue with discussion, for issues with possible bundling please open new issue and ideally prepare example project.
